samples/grpc_reflection.js

```diff
@@ -0,0 +1,21 @@
+import grpc from 'k6/net/grpc';
+import { check } from "k6";
+
+let client = new grpc.Client();
+client.reflect("127.0.0.1:10000", { plaintext: true })
```

Comment:
I agree with @na-- that having this in the init context is problematic. Keep in mind that the init context is evaluated at least 4 times (for getting the `options`, for `setup()`, `teardown()` and `handleSummary()` even if these aren't defined in the script), and then for every created VU. So it's wasteful to have the gRPC connection/reflection happen here multiple times, and it could impact VU initialization time.

To me it would make sense to do this in `setup()`, but since we can't easily share data from there it wouldn't be feasible for distributed execution. Maybe we should prioritize having a per-VU init function (#785), though even in that case it would be wasteful for each VU to do the reflection, and the service definitions (whether loaded statically or reflected) should be shared among VUs anyway. So it seems like we need some kind of `SharedArray` and per-VU init to make this happen, which is not trivial to add.

And from an API perspective, I think reflection should be an implementation detail, so also agree with @na-- that it would make sense for this to be an [optional parameter to the `Client` constructor](https://github.com/k6io/k6/issues/1930#issuecomment-808890461), instead of a separate method that scripts have to call.

Could it be added as an argument to `client.connect()` instead, in which case we could cache the definitions internally and avoid the init context problems? :thinking:

Reply:
That could work as well, but I'd prefer if we avoided the use of `__ITER` for what should be built-in functionality, so I'm for handling that internally in `connect()` if possible. There's no reason to have scripts manually call `getProtoDefinitions()` unless we plan to expose them in JS for some reason.

Reply:
btw since it seems like the constructor doesn't have any parameters currently, while `connect()` does, and since it seems like the very rough consensus is that this should happen during `connect()`, should the option be a `connect()` parameter?

Reply:
I am not very versed in gRPC, but from what I can see:
1. in order to have the reflection we need to connect to the server so it is only natural that it should happen with the `connect` method or after it
2.  if there is a way to detect that the server can give us the proto files I think k6 should just get the, And then have a way to tell it *not* to with an argument on the `connect` method. If it isn't possible I have no opinion on whether it should be an argument or a function call after the connect :shrug: 
3. We can definitely cache *across-VUs* as well as *per-vu* with the same/similar method used in SharedArray or the [xk6-segment extension](https://github.com/MStoykov/xk6-segment/blob/8df18d74cb838a7c16167afec6feabb65ae93af0/pkg/segment/segment.go#L39-L70) ... I do consider this trivial as well ;) . Although probably there should be a way to tell if you don't want to :shrug: maybe in a separate PR. This should remove the whole `__ITER==0` thing and we can probably use the server address as the key (from what I understand)

Reply:
cross-VU sharing of protobuf definitions is definitely something out of scope for this PR

Reply:
I've changed it so that the `client.reflect` function is called at the end of the `connect` function only if `{reflect: true}` is passed in. 

I have used a `sync.Once` to ensure that it is only called once per client